File: tws-graph/tests/fixtures/clojure/Sample.clj
;; Fixture: Sample Clojure file for extractor testing
(ns sample.core
  (:require [clojure.string :as str]
            [clojure.set :refer [union difference]])
  (:import [java.util Date Calendar]
           [java.io File]))

(def app-name "SampleApp")

(defn greet
  "Returns a greeting string"
  [name]
  (str "Hello, " name "!"))

(defn process-data
  [data]
  (let [cleaned (str/trim data)
        items (str/split cleaned #",")]
    (map str/upper-case items)))

(defmacro log-and-do
  [msg & body]
  `(do
     (println (str "[LOG] " ~msg))
     ~@body))

(defn -main
  [& args]
  (let [msg (str/join " " args)]
    (println (greet msg))))

File: tws-graph/tests/fixtures/haskell/Sample.hs
-- Fixture: Sample Haskell file for extractor testing
module SampleApp where

import Data.List (sort, nub)
import qualified Data.Map as Map
import Data.Maybe (fromMaybe)

data Person = Person { personName :: String, personAge :: Int }
  deriving (Show, Eq)

newtype UserId = UserId Int

type Name = String

class Greeting a where
  greet :: a -> String

instance Greeting Person where
  greet p = "Hello, " ++ personName p

sayHello :: Person -> IO ()
sayHello person = do
  let name = personName person
  putStrLn (greet person)

main :: IO ()
main = do
  let alice = Person "Alice" 30
  sayHello alice
